src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1738,26 +1746,68 @@ def client_side_proxy_wrapper(cmd,
+
+            for i in range(len(creds_list)):
+                creds_obj = creds_list[i]
+
+                if key in creds_obj and creds_obj[key] == user_name:
+                    creds = creds_obj[key2]
+                    break
+            if creds == '':
+                telemetry.set_exception(exception='Credentials of user not found.', fault_type=consts.Creds_NotFound_Fault_Type,
+                                        summary='Unable to find creds of user while using ADAL')
+                raise UnclassifiedUserFault("Credentials of user not found.")
         else:
-            key = 'servicePrincipalId'
-            key2 = 'accessToken'
-
-        for i in range(len(creds_list)):
-            creds_obj = creds_list[i]
-
-            if key in creds_obj and creds_obj[key] == user_name:
-                creds = creds_obj[key2]
-                break
-
-        if creds == '':
-            telemetry.set_exception(exception='Credentials of user not found.', fault_type=consts.Creds_NotFound_Fault_Type,
-                                    summary='Unable to find creds of user')
-            raise UnclassifiedUserFault("Credentials of user not found.")
-
+            if user_type == "user":
+                response_user_objectid = az_cli("ad signed-in-user show --query objectId -o tsv")
+                token_cache_location = os.path.expanduser(os.path.join('~', msal_token_cache_user))
+                try:
+                   if operating_system == 'Windows':
+                        persistence = msal_extensions.FilePersistenceWithDataProtection(token_cache_location)
+                   elif operating_system == 'Darwin':
+                        persistence = msal_extensions.KeychainPersistence(token_cache_location)
+                   else:
+                        persistence = msal_extensions.FilePersistence(token_cache_location)
+                   token_cache = msal_extensions.PersistedTokenCache(persistence)      
+                except  Exception as e:
+                    telemetry.set_exception(exception=e, fault_type=consts.MSAL_cache_not_retrieved_Fault_Type,
+                                    summary='Unable to retrieve MSAL cache')
+                    raise CLIInternalError("Failed to authenticate with the cluster")
+                token_cache._reload_if_necessary()
+                home_account_id = response_user_objectid + "." + tenantId
+                owned_by_home_account = {"home_account_id": home_account_id}
+                creds_list = token_cache.find(PersistedTokenCache.CredentialType.REFRESH_TOKEN, query=owned_by_home_account)
+                creds = creds_list[0]['secret']
+            else:
+                token_cache_location = os.path.expanduser(os.path.join('~', msal_token_cache_spn))
+                try:
+                   if operating_system == 'Windows':
+                        persistence = msal_extensions.FilePersistenceWithDataProtection(token_cache_location)
```

Comment:
You really should not be doing this. MSAL's internal cache is not a public artifact and may change without notice. 

In fact, this is changing as we speak - MSAL py will use the Windows broker (WAM). Brokers don't even return RT and cache is going to be stored elsewhere (not shared anymore).